quickjs.c

```diff
@@ -3874,6 +3874,20 @@ static JSValue string_buffer_end(StringBuffer *s)
     return JS_MKPTR(JS_TAG_STRING, str);
 }
 
+JSValue JS_NewStringWLen(JSContext *ctx, size_t buf_len)
+{
+    JSString *str;
+    if (buf_len <= 0) {
+        return JS_AtomToString(ctx, JS_ATOM_empty_string);
+    }
+    str = js_alloc_string_rt(ctx->rt, buf_len, 0);
+    if (unlikely(!str)){
+        JS_ThrowOutOfMemory(ctx);
+        return JS_EXCEPTION;
+    }
+    memset(str->u.str8, 0, buf_len+1);
+    return JS_MKPTR(JS_TAG_STRING, str);
+}
```

Comment:
Instead of a new API, I would modify `JS_NewStringLen` to accept a null pointer and initialize the allocated string with null bytes using this:
```
    if (buf == NULL) {
        JSString *str;

        if (buf_len <= 0) {
            return JS_AtomToString(ctx, JS_ATOM_empty_string);
        }
        str = js_alloc_string(ctx, buf_len, 0);
        if (!str)
            return JS_EXCEPTION;
        memset(str->u.str8, buf, buf_len + 1);
        return JS_MKPTR(JS_TAG_STRING, str);
    }
```